File: Banking-Kata/src/test/java/banking/persistence/FakeRepository.java
package banking.persistence;

import banking.model.Identifiable;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public class FakeRepository<T extends Identifiable> implements Repository<T> {

  public Map<UUID, T> items = new HashMap<>();

  @Override
  public Optional<T> findOne(final UUID id) {
    return Optional.ofNullable(items.get(id));
  }

  @Override
  public void save(T item) {
    items.put(item.getId(), item);
  }
  
  public List<T> allItems() {
    return new ArrayList<>(items.values());
  }
  
  public void clear() {
    items.clear();
  }

}

File: 07-Banking-Kata/csharp-dotnetcore2/BankingKataAPI.Test/Persistence/FakeRepository.cs
using System;
using System.Collections.Generic;
using System.Text;
using BankingKataAPI.Models;
using BankingKataAPI.Persistence;

namespace BankingKataAPI.Test.Persistence
{
    public class FakeRepository<T> : IRepository<T> where T : class, Identifiable
    {
        private readonly Dictionary<Guid, T> _items;

        public FakeRepository()
        {
            _items = new Dictionary<Guid, T>();
        }

        public T FindOne(Guid id)
        {
            return _items.ContainsKey(id) ? _items[id] : null;
        }

        public void Save(T item)
        {
            _items[item.GetId()] = item;
        }
    }
}
